src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGeneratorTest.java

```diff
@@ -514,17 +515,70 @@ public void testTextBlocks() throws Exception {
-        assertEquals(expectedTestOne, actualTestOne,
-                "Generated queries do not match expected ones");
+        assertWithMessage("Generated queries do not match expected ones")
+                .that(expected).isEqualTo(actual);
+    }
+
+    @Test
+    public void testTextBlocksWithNewLine() throws Exception {
+        final File testFile = new File(getNonCompilablePath(
+                "InputXpathQueryGeneratorTextBlockNewLine.java"));
+        final FileText testFileText = new FileText(testFile,
+                StandardCharsets.UTF_8.name());
+        final DetailAST detailAst =
+                JavaParser.parseFile(testFile, JavaParser.Options.WITHOUT_COMMENTS);
+        final int tabWidth = 8;
+
+        final int lineNumber = 6;
+        final int columnNumber = 25;
+        final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(detailAst,
+                lineNumber, columnNumber, testFileText, tabWidth);
+        final List<String> actual = queryGenerator.generate();
+        final List<String> expected = Collections.singletonList(
+                "/CLASS_DEF[./IDENT[@text='InputXpathQueryGeneratorTextBlockNewLine']]/OBJBLOCK/"
+                        + "VARIABLE_DEF[./IDENT[@text='testOne']]/ASSIGN/EXPR/"
+                        + "TEXT_BLOCK_LITERAL_BEGIN/TEXT_BLOCK_CONTENT[@text='\\n        "
+                        + "&amp;1line\\n\\n        &gt;2line\\n        &lt;3line\\n        ']"
+        );
+        assertWithMessage("Generated queries do not match expected ones")
+                .that(expected).isEqualTo(actual);
     }
+
+    @Test
+    public void testTextBlocksWithNewCrlf() throws Exception {
+        final File testFile = new File(getNonCompilablePath(
+                "InputXpathQueryGeneratorTextBlockCrlf.java"));
+        final FileText testFileText = new FileText(testFile,
+                StandardCharsets.UTF_8.name());
+        final DetailAST detailAst =
+                JavaParser.parseFile(testFile, JavaParser.Options.WITHOUT_COMMENTS);
+        final int tabWidth = 8;
+
+        final int lineNumber = 6;
+        final int columnNumber = 25;
+        final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(detailAst,
+                lineNumber, columnNumber, testFileText, tabWidth);
+        final List<String> actual = queryGenerator.generate();
+        final List<String> expected = Collections.singletonList(
+                "/CLASS_DEF[./IDENT[@text='InputXpathQueryGeneratorTextBlockCrlf']]/OBJBLOCK/"
+                        + "VARIABLE_DEF[./IDENT[@text='testOne']]/ASSIGN/EXPR/"
+                        + "TEXT_BLOCK_LITERAL_BEGIN/TEXT_BLOCK_CONTENT[@text='\\r\\n        "
+                        + "&amp;1line\\r\\n\\r\\n        &gt;2line\\r\\n        &lt;3line\\r\\n"
+                        + "        ']"
+        );
+        assertWithMessage("Generated queries do not match expected ones")
+                .that(expected).isEqualTo(actual);
+    }
+
```

Comment:
@AkMo3 good work! I have only one comment

Please one more test in `XpathMapperTest` to validate that you can query `TEXT_BLOCK_CONTENT` nodes by its content, see example for `NUM_INT`

Reply:
Done